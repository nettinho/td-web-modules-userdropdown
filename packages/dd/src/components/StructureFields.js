import _ from "lodash/fp";
import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { Table, Divider } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import StructureFieldRow from "./StructureFieldRow";

const hasLineage = _.any(_.prop("external_id"));
const hasConceptsLink = fields =>
  !_.all(e => _.isEmpty(_.property("bc_related")(e)))(fields);

export const StructureFields = ({ fields, ...rest }) => (
  <React.Fragment>
    <Divider hidden />
    <Table celled striped compact selectable attached="bottom">
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell
            content={<FormattedMessage id="structure.field.name" />}
          />
          <Table.HeaderCell
            content={<FormattedMessage id="structure.field.type" />}
          />
          <Table.HeaderCell
            content={<FormattedMessage id="structure.field.precision" />}
          />
          <Table.HeaderCell
            content={<FormattedMessage id="structure.field.nullable" />}
          />
          <Table.HeaderCell
            content={<FormattedMessage id="structure.field.description" />}
          />
          {hasLineage(fields) && (
            <Table.HeaderCell
              content={<FormattedMessage id="structure.field.external_id" />}
            />
          )}
          {hasConceptsLink(fields) && (
            <Table.HeaderCell
              content={<FormattedMessage id="structure.field.concepts" />}
            />
          )}
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {fields.map((f, i) => (
          <StructureFieldRow
            key={i}
            lineageVisible={hasLineage(fields)}
            conceptsLinkVisible={hasConceptsLink(fields)}
            {...f}
            {...rest}
          />
        ))}
      </Table.Body>
    </Table>
  </React.Fragment>
);

StructureFields.propTypes = {
  fields: PropTypes.array
};

const mapStateToProps = ({ structure }) => ({
  data_structure_id: structure.id,
  fields: structure.data_fields
});

export default connect(mapStateToProps)(StructureFields);
