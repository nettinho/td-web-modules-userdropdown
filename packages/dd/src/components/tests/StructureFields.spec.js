import React from "react";
import { shallow } from "enzyme";
import { StructureFields } from "../StructureFields";

describe("<StructureFields />", () => {
  it("matches the latest snapshot", () => {
    const structure = {
      fields: [
        {
          id: 123,
          name: "field1",
          description: "desc1",
          type: "varchar",
          nullable: true,
          precision: "22"
        },
        {
          id: 122,
          name: "field2",
          description: "desc2",
          type: "varchar",
          nullable: true,
          precision: "22"
        }
      ]
    };

    const wrapper = shallow(<StructureFields {...structure} />);
    expect(wrapper).toMatchSnapshot();
  });
});
