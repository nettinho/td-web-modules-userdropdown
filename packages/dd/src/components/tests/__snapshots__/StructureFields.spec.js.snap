// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StructureFields /> matches the latest snapshot 1`] = `
<React.Fragment>
  <Divider
    hidden={true}
  />
  <Table
    as="table"
    attached="bottom"
    celled={true}
    compact={true}
    selectable={true}
    striped={true}
  >
    <TableHeader
      as="thead"
    >
      <TableRow
        as="tr"
        cellAs="td"
      >
        <TableHeaderCell
          as="th"
          content={
            <FormattedMessage
              id="structure.field.name"
              values={Object {}}
            />
          }
        />
        <TableHeaderCell
          as="th"
          content={
            <FormattedMessage
              id="structure.field.type"
              values={Object {}}
            />
          }
        />
        <TableHeaderCell
          as="th"
          content={
            <FormattedMessage
              id="structure.field.precision"
              values={Object {}}
            />
          }
        />
        <TableHeaderCell
          as="th"
          content={
            <FormattedMessage
              id="structure.field.nullable"
              values={Object {}}
            />
          }
        />
        <TableHeaderCell
          as="th"
          content={
            <FormattedMessage
              id="structure.field.description"
              values={Object {}}
            />
          }
        />
      </TableRow>
    </TableHeader>
    <TableBody
      as="tbody"
    >
      <withRouter(InjectIntl(StructureFieldRow))
        conceptsLinkVisible={false}
        description="desc1"
        id={123}
        key="0"
        lineageVisible={false}
        name="field1"
        nullable={true}
        precision="22"
        type="varchar"
      />
      <withRouter(InjectIntl(StructureFieldRow))
        conceptsLinkVisible={false}
        description="desc2"
        id={122}
        key="1"
        lineageVisible={false}
        name="field2"
        nullable={true}
        precision="22"
        type="varchar"
      />
    </TableBody>
  </Table>
</React.Fragment>
`;
